jerry-core/ecma/base/ecma-helpers-number.cpp

```diff
@@ -916,6 +916,60 @@ ecma_number_exp (ecma_number_t num) /**< valid finite number */
   return sum;
 } /* ecma_number_exp */
 
+
+/**
+ * Helper for preprocessing, convert ecma_number_t to uint_32
+ *
+ */
+void
+ecma_number_to_u32 (ecma_number_t num, bool *p_infinity, bool *p_sign,
+    uint64_t *p_cast_to_u64)
+{
+  const uint32_t fraction_pos = 0;
+  const uint32_t biased_exp_pos = fraction_pos + ECMA_NUMBER_FRACTION_WIDTH;
+  const uint32_t sign_pos = ECMA_NUMBER_FRACTION_WIDTH + ECMA_NUMBER_BIASED_EXP_WIDTH;
+
+  union
+  {
+    uint64_t u64_value;
+    ecma_number_t float_value;
+  } u;
+
+  u.float_value = num;
+  uint64_t packed_value = u.u64_value;
+
+  uint32_t biased_exp = (uint32_t) (((packed_value) & ~(1ull << sign_pos)) >> biased_exp_pos);
+  uint64_t fraction = (packed_value & ((1ull << ECMA_NUMBER_FRACTION_WIDTH) - 1));
+
+  *p_infinity = ((biased_exp  == (1u << ECMA_NUMBER_BIASED_EXP_WIDTH) - 1)
+      && (fraction == 0));
```

Comment:
Check infinity here is to avoid call ecma_number_is_infinity() from ecma_number_to_uint32().It is not necessary to unpack the floating point num more than one time. All I am trying to do is , unpack a floating point one time, get biased exp, sign and fraction, then we can do everything.